### PR TITLE
added rfft

### LINKED ISSUE #5501: rfft

### PR DESCRIPTION
**Pull request by Viditagarwal7479**
Close #5501

### CODE REVIEW

**Review by MarShaikh — CHANGES_REQUESTED** [review]
Can you update your PR to not add changes to the `test_array_api`?

**Comment by Viditagarwal7479** [comment]
Hey,
the extensions.py files have been removed from all the folders, and a new directory extension has been created in a recent change which took place in this PR https://github.com/unifyai/ivy/pull/5970

So in which file of the extensions directory shall I add ```rfft``` function?

Thanks
Vidit

**Comment by MarShaikh** [comment]
> Hey, the extensions.py files have been removed from all the folders, and a new directory extension has been created in a recent change which took place in this PR #5970
> 
> So in which file of the extensions directory shall I add `rfft` function?
> 
> Thanks Vidit

You can add it to the `ivy/functional/ivy/extensions

> Hey, the extensions.py files have been removed from all the folders, and a new directory extension has been created in a recent change which took place in this PR #5970
> 
> So in which file of the extensions directory shall I add `rfft` function?
> 
> Thanks Vidit

You can add it to `ivy.functional.extensions.layers`!

File: ivy/array/extensions/layers.py [hunk 1]
```diff
@@ -262,3 +262,45 @@ def pad(
...
+        --------
+        >>> x = ivy.array([0, 1, 2, 3])
+        >>> ivy.rfft(x)
+        ivy.array([ 6.+0.j, -2.+2.j, -2.+0.j])
+        >>> ivy.rfft(x, norm='ortho')
+        ivy.array([ 3.+0.j, -1.+1.j, -1.+0.j])
+        >>> ivy.rfft(x, norm='forward')
+        ivy.array([ 1.5000+0.0000j, -0.5000+0.5000j, -0.5000+0.0000j])
+        """
+        return ivy.rfft(input, n, norm, out=out)
```
Comment (MarShaikh) [inline ivy/array/extensions/layers.py:306]: The way you have written this array method isn't correct, please check the other examples in file!

File: ivy/array/extensions/layers.py [hunk 2]
```diff
@@ -262,3 +262,45 @@ def pad(
...
+
+        Returns
+        -------
+        ret
+            The output array.
+
+        Examples
+        --------
+        >>> x = ivy.array([0, 1, 2, 3])
+        >>> ivy.rfft(x)
```
Comment (MarShaikh) [inline ivy/array/extensions/layers.py:299]: Also, please refer to the docs to see the correct way of calling array instance methods!

**Review by MarShaikh — CHANGES_REQUESTED** [review]

**Review by MarShaikh — CHANGES_REQUESTED** [review]
Finally if you see the test-extensions in the CI, it is also failing, can you check that as well!

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**